sdks/python/apache_beam/io/range_trackers.py

```diff
@@ -285,6 +309,14 @@ def fraction_consumed(self):
                        ' consumed due to positions being opaque strings'
                        ' that are interpreted by the service')
 
+  def split_points(self):
+    with self._lock:
+      splits_points_consumed = (
+          0 if self._split_points_seen <= 1 else (self._split_points_seen - 1))
+
+      return (splits_points_consumed,
+              iobase.RangeTracker.SPLIT_POINTS_UNKNOWN)
```

Comment:
One of the most important cases is where there are few large (straggling) keys, in which case you can return that there are no more split points iff the endpoint is the successor to this key.